### ISSUE TITLE
App-related commands do not work in dev environment

### ISSUE THREAD

**Issue by jglick** [jenkins-x/jx#2971]
### Summary

Working on an app similar to `jx-app-jacoco`, I was interested in adding it to the `dev` environment (i.e., `jx` namespace) in a test cluster, so that it would be applied to builds of sample projects. It turned out the `jx` subcommands for managing apps were not usable and I had to use Helm directly.

It seems that [deployments in the `dev` environment are ignored](https://github.com/jenkins-x/jx/blob/24e0a628dda703737026abd0f22ec6a132bfbe59/pkg/jx/cmd/get_applications.go#L254) for whatever reason.

Also it seems wrong that the code is checking for `Deployments`. Is it illegal to have an app whose chart does not include a `Deployment`, perhaps because it is doing something else (adding a batch job, registering a CR with some operator, …)? Or which includes multiple `Deployment`s?

### Steps to reproduce the behavior

Not sure how to reproduce from scratch.

### Expected behavior

Perhaps I misunderstand what apps are or how they are expected to work, but I assumed that

```sh
jx add app APPNAME
```

if it worked would then result in

```sh
jx get apps
```

listing the app in the currently installed version and

```sh
jx upgrade app APPNAME
```

would switch to a new version.

### Actual behavior

In a `jx shell` in the `jx` namespace, I tried

```sh
jx add app APPNAME --repository=https://AUTH@chartmuseum.DOMAIN
```

(The URL is from `helm repo list`. Without `--repository` the command complains that it cannot find the app.)

That command succeeded, and indeed `helm ls` displays the release as `DEPLOYED` into the `jx` namespace, and `kubectl get deployments` shows it running. Yet

```sh
jx get apps
```

does not list it; only a couple of things that look like test projects I cannot identify (not in `helm ls`).

Later I wanted to upgrade the app to a new release.

```sh
jx upgrade apps APPNAME
```

succeeds with no output and does nothing that I can see. Ditto with `--repository=…`. The only thing that worked was

```sh
helm upgrade APPNAME REPONAME/APPNAME
```

(with `REPONAME` again being taken from `helm repo list`).

Oh, and `kubectl get apps.jenkins.io` lists a couple of _other_ things (which `helm ls` does also acknowledge and which have running `deployments`), but not the app I tried to add.

### Jx version

The output of `jx version` is:

```
NAME               VERSION
jx                 1.3.817
jenkins x platform 0.0.3321
Kubernetes cluster v1.11.5-gke.5
kubectl            v1.13.2
helm client        v2.12.3+geecf22f
helm server        v2.12.2+g7d2b0c7
git                git version 2.19.1
Operating System   Ubuntu 18.10
```

### Jenkins type

- [ ] Classic Jenkins
- [X] Serverless Jenkins

### Kubernetes cluster

```yaml
apiVersion: jenkins.io/v1
kind: Environment
metadata:
  labels:
    env: dev
    team: jx
  name: dev
  namespace: jx
  # …
spec:
  kind: Development
  label: Development
  namespace: jx
  previewGitInfo:
    user: {}
  promotionStrategy: Never
  source: {}
  teamSettings:
    buildPackName: classic-workloads
    buildPackRef: master
    buildPackUrl: https://github.com/jenkins-x-buildpacks/jenkins-x-classic.git
    gitServer: https://github.com
    kubeProvider: gke
    pipelineUsername: …
    promotionEngine: Prow
    quickstartLocations:
    - excludes:
      - WIP-*
      gitKind: github
      gitUrl: https://github.com
      includes:
      - '*'
      owner: jenkins-x-quickstarts
    storageLocations:
    - classifier: default
      gitBranch: gh-pages
      gitUrl: https://github.com/…
    useGitOps: true
  webHookEngine: Prow
status: {}
```

**Comment by pmuir** [comment]
Some of the confusion here is the `jx get apps` is not yet implemented - it's still actually aliased to `jx get applications` which displays your deployed apps (hence seeing the `Deployment`). 

https://github.com/jenkins-x/jx/issues/2976

> (The URL is from helm repo list. Without --repository the command complains that it cannot find the app.)

Yes. You need to put the `--repository` if you aren't publishing to the default repo, I need to find a better UX here. You can also use `--username` and `--password` to avoid having to build them into the URL. Again, I need a better UX to detect if a username and password are required and prompt for them.
> Later I wanted to upgrade the app to a new release.
> jx upgrade apps APPNAME

Bug, created https://github.com/jenkins-x/jx/issues/2977

> helm upgrade APPNAME REPONAME/APPNAME
> (with REPONAME again being taken from helm repo list).

Another workaround to do it is just to rerun `jx add apps`

> Oh, and kubectl get apps.jenkins.io lists a couple of other things (which helm ls does also acknowledge and which have running deployments), but not the app I tried to add.

Right now I haven't added a synthetic entry here if the app doesn't bundle the app CRD. https://github.com/jenkins-x/jx/issues/2978

**Comment by igdianov** [comment]
@pmuir I am glad I stumbled on this issue because I have tried to use `jx add app` recently and found the experience confusing.

Basically, I was expecting `jx add app` to ask to choose from existing GitOps environment (i.e. via survey or `--env` argument) where to install the app using Helm chart from specified repository and then promote it there via GitOps using custom values. 

Instead, Jx always chooses `dev` environment mapped to `jx` namespace using hardcoded `dev=dev` selector in the code. There is no default Git Url mapped to Jx environment, so it then simply runs `helm upgrade --install` in `jx` namespace without GitOps promotion. If I do `jx edit env -n jx` environment to add Git repo to it, Jx tries to use Vault and reports error because Vault is not installed. 

If not using Vault, It would be useful to allow app with custom values to be added to environment's values.yaml under app name or alias key, so it will work for folks like me without running Vault.
